File: 이진영/Greedy/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => a - b);
    
    let count = 0;
    let l = 0;
    let r = people.length - 1;
    
    while (l < r) {
        if (people[l] + people[r] <= limit) l++;
        r--;
        
        count++;
    }
    if (l === r) count++;
    
    return count;
}

File: 김채린/Greedy/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        List list = new ArrayList<Integer>();
        for(int n:people) {
            list.add(n);
        }
        int count=0;
        list.sort(Comparator.reverseOrder());
        for (int i=0; i<list.size(); i++){
            int total = 0;
            total += (int) list.get(i);
            if (list.size()>1&&total+(int)list.get(list.size()-1)<=limit){
                list.remove(list.size()-1);
            }
            count++;
        }
        return count;
    }
}
